### DISCUSSION TITLE
Setting osd_memory_target in ceph does not take effect after configuring request memory in cluster.yaml

### DISCUSSION THREAD

**Discussion by mc956759** [rook/rook#12277] [Q&A] [answered]
As the title described, I configured osd-ssd request memory in cluster.yaml.
But while cluster running, setting osd_memory_target in ceph does not take effect! (use ceph config set osd osd_memory_target 21474836480)
What happend while osd tune memoy in rook-ceph?
And how can I change my osds' osd_memory_target ? I know I can modify the cluster.yaml, and apply to k8s cluster. But will the OSD PODs restart at the same time rather than one by one?
Please help me !
Thanks a lot!

 osd-ssd:
      limits:
        cpu: "10000m"
        memory: "20Gi"
      requests:
        cpu: "2000m"
        memory: "10Gi"

**Comment by sp98** [comment]
The pods will restart one by one.

**Reply by mc956759** [reply]
But I found that pods on one node will restart at the same time. Maybe is there something wrong with the configuration?

**Reply by sp98** [reply]
do you mean osds pods on a node restart at the same time?

**Reply by mc956759** [reply]
yes, osds pods on a node.

**Reply by sp98** [reply]
ok. What's the concern with osd pods on a node restarting at the same time?

**Reply by mc956759** [reply]
I was worried about data reliability. Since one osd have 7T data, I didn't know how long it need to start.
In fact it's very fast, and the cluster works will during osds restarting.
Thanks a lot!

**Comment by mc956759** [comment]
I found that I can use "ceph tell osd.0 config set osd_target_memory xxxx" to modify this configuration. But it's just runtime, which means it will be reverted after I restart the pod.

**Reply by sp98** [reply, marked as answer]
Yes, you are right. If you have specified the limits on the cluster.yaml, the osds will use those limits and override any change that was done using ` ceph config set osd osd_memory_target VALUE`. 
The recommended way is to set the limits in the cluster.yaml.  

If you haven't already, then I would suggest going over this [issue](https://github.com/rook/rook/issues/9110) once.

**Reply by mc956759** [reply]
Thanks! I have found the method of "ceph tell osd.0 config set osd_target_memory xxxx" from this issue.
